app/views/finance/provider/settings/show.html.slim

```diff
@@ -1,135 +1,114 @@
+      div class="pf-c-card__title" Credit card gateway
+      div class="pf-c-card__body"
+        = semantic_form_for @account, url: admin_account_payment_gateway_path,
+                                      html: { id: 'payment-gateway-form', class: 'pf-c-form pf-m-limit-width' } do |form|
+          = form.input :payment_gateway_type, as: :patternfly_select,
+                                              member_value: :type,
+                                              collection: @payment_gateways,
+                                              include_blank: 'Choose one'
+
+          = form.fields_for :payment_gateway_options do |fields|
+            - @payment_gateways.each do |payment_gateway|
+              - if payment_gateway.fields.present?
+                - is_current = payment_gateway.type == @account.payment_gateway_type
+                section class="pf-c-form__section payment_gateway_options #{is_current ? '' : 'hidden'}" id="payment_gateway_#{payment_gateway.type}" role="group"
+                  - payment_gateway.fields.each do |name, label|
                     - if payment_gateway.boolean_field_keys.include?(name)
-                      = fields.label name, for: ['account', 'payment', payment_gateway.type, name].join('_') do
-                        - if enabled
-                          = fields.check_box name, checked: @account.payment_gateway_options[name].present?, id: ['account', 'payment', payment_gateway.type, name].join('_')
-                        - else
-                          = fields.check_box name, checked: false, disabled: true, id: ['account', 'payment', payment_gateway.type, name].join('_')
-
-                        = label
-                      p.inline-hints Make sure your Braintree account has 3D Secure enabled before you enable it here as otherwise payments initiated from the developer portal will fail.
-
+                      - if is_current
+                        = fields.input name, as: :patternfly_checkbox,
+                                             id: ['account', 'payment', payment_gateway.type, name].join('_'),
+                                             input_html: { checked: @account.payment_gateway_options[name].present? }
+                      - else
+                        = fields.input name, as: :patternfly_checkbox,
+                                             label: label,
+                                             checked: false,
+                                             id: ['account', 'payment', payment_gateway.type, name].join('_')
                     - else
-                      = fields.label name, label
-                      - if enabled
-                        = fields.text_field name, value: @account.payment_gateway_options[name]
+                      - if is_current
+                        = fields.input name, as: :patternfly_input,
+                                             input_html: { value: @account.payment_gateway_options[name] }
```

Comment:
```suggestion
                        = fields.input name, as: :patternfly_input,
                                             label: label,
                                             input_html: { value: @account.payment_gateway_options[name] }
```

Reply:
Otherwise, proper label is not applied:
![image](https://github.com/3scale/porta/assets/1270328/43618136-72c5-426e-b869-4dd33d63922a)

vs 
![image](https://github.com/3scale/porta/assets/1270328/f56cf1c7-8411-4e15-bd1f-794b3cbf81e2)